packages/modal/src/Modal/Modal.tsx

```diff
@@ -11,89 +13,183 @@ import { Box, Stack } from "@telegraph/layout";
 import { AnimatePresence, motion } from "framer-motion";
 import React from "react";
 
+import { DismissableWrapper } from "./Modal.helpers";
+import { useModalStacking } from "./ModalStacking";
+
 type RootProps = Omit<
   React.ComponentPropsWithoutRef<typeof Dialog.Root>,
   "modal"
 > &
   TgphComponentProps<typeof Stack> & {
     a11yTitle: string;
     a11yDescription?: string;
+    layer?: number;
   };
 
 const Root = ({
-  defaultOpen,
+  defaultOpen: defaultOpenProp,
+  open: openProp,
+  onOpenChange: onOpenChangeProp,
+  ...props
+}: RootProps) => {
+  const [open, onOpenChange] = useControllableState({
+    prop: openProp,
+    onChange: onOpenChangeProp,
+    defaultProp: defaultOpenProp,
+  });
+
+  // Prevent rendering anything within the modal if it is not open
+  if (!open) return;
+
+  return <RootComponent open={open} onOpenChange={onOpenChange} {...props} />;
+};
```

Comment:
Added a wrapping component for `Modal.Root` so that we only initiate any of the contents of `Modal.Root` when the modal is actually open.